### PR TITLE
Changing the text color of tracks taking the background color into account

### PR DESCRIPTION
**Pull request by Ana06**
Issue #1033 

Before:

image

Now:

image

I didn't change it in the schedule as it is being changed and there are two pull-request open. I'll change afterwards.

### CODE REVIEW

**Comment by hennevogel** [comment]
We are using [tinycolor](https://github.com/bgrins/TinyColor) already in the splash page to calculate colors. Can't we use this instead of re-inventing it ourselves?

**Comment by Ana06** [comment]
We could, but tinycolor `mostReadable` function selects black or white depending on what of them has more readability, and readability is given by:

``` Javascript
tinycolor.readability=function(color1,color2){
    var c1=tinycolor(color1),c2=tinycolor(color2);
    return(Math.max(c1.getLuminance(),c2.getLuminance())+.05)/(Math.min(c1.getLuminance(),c2.getLuminance())+.05)
}
```

So, it is not exactly the same and will select a different color for some cases. In the following image you can see the difference for the colors I used before. The first column is my option and the second is tinycolor option. As you can see, they are different for two colors. In my opinion, and as we are talking about color it is probably a matter of taste, my option looks better.

image


Also, I don't like the javascript needed to use tinycolor for that. That the javascript I wrote for this:

``` Javascript
$(function () {
  $(document).ready(function() {
    var elements = document.getElementsByClassName("track");
    for(var i=0; i < elements.length; i++) {
        var background = elements[i].style.backgroundColor
        var color = tinycolor.mostReadable(background,["#000", "#fff"]).toHexString();
        elements[i].style.color = color;
    }
  });
});
```

And we will have a dependency of a library to use one function that is easily to implement.

In short, I'll use my first implementation. :wink:

**Comment by hennevogel** [comment]
Her we go :-)
